File: PROGRAMMERS/Lv1/모의고사.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    
    int one_arr[5] = {1, 2, 3, 4, 5};
    int two_arr[8] = {2, 1, 2, 3, 2, 4, 2, 5};
    int three_arr[10] = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    
    int one = 0;
    int two = 0;
    int three = 0;
    
    for(int i=0; i < answers.size(); i++){
        if(answers[i] == one_arr[i%5]) one++;
        if(answers[i] == two_arr[i%8]) two++;
        if(answers[i] == three_arr[i%10]) three++;
    }
    
    vector<pair<int, int>> v;
    v.push_back({one, 1});
    v.push_back({two, 2});
    v.push_back({three, 3});

    sort(v.begin(), v.end());
    int max = v[2].first;
    
    for(int i=0; i<3; i++){
        if(v[i].first == max) answer.push_back(v[i].second);
    }
    return answer;
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    
    p1 = [1, 2, 3, 4, 5]
    p2 = [2, 1, 2, 3, 2, 4, 2, 5]
    p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    cnt = [0] * 3
    
    # p1
    for i in range(len(answers)):
        if(answers[i] == p1[i%len(p1)]): cnt[0] += 1
        
    # p2
    for i in range(len(answers)):
        if(answers[i] == p2[i%len(p2)]): cnt[1] += 1
        
    # p2
    for i in range(len(answers)):
        if(answers[i] == p3[i%len(p3)]): cnt[2] += 1
    
    max_size = max(cnt)
    
    for i in range(len(cnt)):
        if(cnt[i] == max_size): 
            answer.append(i+1)
    
    
    return answer